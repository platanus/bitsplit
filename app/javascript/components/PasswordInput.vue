<template>
  <div class="flex justify-end flex-row">
    <input
      class="txt-input rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
      :id="fieldId"
      :type="inputType"
      :placeholder="fieldPlaceholder"
      :name="fieldName"
      :value="value"
      @input="$emit('update', $event.target.value)"
    >
    <img
      @click="swapPassword()"
      :src="noVisibility"
      class="absolute transform translate-y-3 -translate-x-4"
      align="top"
    >
  </div>
</template>
<script>

import noVis from '../assets/no-visibility.svg';

export default {
  name: 'PasswordForm',
  data() {
    return {
      inputType: 'password',
      noVisibility: noVis,
    };
  },
  props: {
    value: String,
    fieldId: String,
    fieldName: String,
    fieldPlaceholder: {
      default: '****************',
      type: String,
    },
  },
  model: {
    prop: 'value',
    event: 'update',
  },
  methods: {
    swapPassword() {
      this.inputType = this.inputType === 'password' 
                        ? 'text' 
                        : 'password'
    },
  }
};
</script>
