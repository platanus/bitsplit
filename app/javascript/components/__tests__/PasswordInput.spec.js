import { shallowMount } from '@vue/test-utils'
import PasswordInput from '../PasswordInput'

describe('PasswordInput', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(PasswordInput, {
      propsData: {
        fieldName: 'password',
        fieldId: 'password'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
