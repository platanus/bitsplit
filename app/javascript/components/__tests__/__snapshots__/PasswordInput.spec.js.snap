// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput matches snapshot 1`] = `<div class="flex justify-end flex-row"><input id="password" type="password" placeholder="****************" name="password" class="txt-input rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"> <img src="[object Object]" align="top" class="absolute transform translate-y-3 -translate-x-4"></div>`;
